docs/iris/src/whatsnew/aggregate_directory.py

```diff
@@ -176,9 +260,16 @@ def run_compilation(release):
         for string_line in compiled_text:
             output_object.write(string_line)
 
+
 if __name__ == '__main__':
     PARSER = argparse.ArgumentParser()
+    _DUMMY_VERSION = '9999.9999'
     PARSER.add_argument("release", help="Release number to be compiled",
+                        nargs='?', default=_DUMMY_VERSION,
                         type=version.StrictVersion)
     ARGUMENTS = PARSER.parse_args()
-    run_compilation(ARGUMENTS.release)
+    release = ARGUMENTS.release
+    if release == _DUMMY_VERSION:
+        release = get_latest_release()
+    print('Building release document for release "{!s}"'.format(release))
```

Comment:
There's a lot I'm not keen on in the control/calling structure of this module, but I didn't want to  rewrite more than I had to, as I felt the reworks were getting unnecessarily involved and I might encounter opposition to spending so much effort on this non-functional code (i.e. including the reviewers' !).

If I were to review + refactor the whole thing, I would certainly want to rationalise some more things, like the way directory + release are passed around.

Even with this much refactoring and testing, I was already itching to add written tests for this, but I feared that might also be judged excessive !